python_modules/libraries/dagster-components/dagster_components/resolved/context.py

```diff
@@ -23,6 +25,25 @@ def env_scope(key: str) -> Optional[str]:
     return os.environ.get(key)
 
 
+def defs_scope(defs_submodule: str) -> Definitions:
```

Comment:
I think we should return a new object whose API is optimized for the templating use case, looking ahead to an world where its properties are themselves annotated with Resolved annotations for introspection by tooling.